protocol/client.go

```diff
@@ -243,31 +263,32 @@ func (client *syncPeerClient) startNewBlockProcess() {
 
 // startPeerEventProcess starts subscribing peer connection change events and process them
 func (client *syncPeerClient) startPeerEventProcess() {
-	peerEventCh, err := client.network.SubscribeCh(context.Background())
+	peerEventCh, err := client.network.SubscribeCh(client.ctx)
 	if err != nil {
 		client.logger.Error("failed to subscribe", "err", err)
 
 		return
 	}
 
+	defer close(client.peerConnectionUpdateCh)
+
 	for e := range peerEventCh {
-		if e.Type == event.PeerConnected || e.Type == event.PeerDisconnected {
-			if client.isClosed.Load() {
-				client.logger.Debug("client is closed, ignore peer connection update", "peer", e.PeerID, "type", e.Type)
+		if client.isClosed.Load() {
+			client.logger.Debug("client is closed, ignore peer connection update", "peer", e.PeerID, "type", e.Type)
+
+			return
+		}
 
+		if e.Type == event.PeerConnected || e.Type == event.PeerDisconnected {
+			select {
+			case <-client.ctx.Done():
 				return
+			case client.peerConnectionUpdateCh <- e:
 			}
-
-			client.peerConnectionUpdateCh <- e
 		}
 	}
 }
 
-// CloseStream closes stream
-func (client *syncPeerClient) CloseStream(peerID peer.ID) error {
```

Comment:
"CloseStream" only closing when the connection is disconnected, and should not be handed over to the user. This design can reduce the complexity of connection management